build-tools/packages/build-cli/src/library/release.ts

```diff
@@ -111,27 +117,35 @@ interface PackageTildeRange {
  * "full" corresponds to the {@link ReleaseReport} interface. It contains a lot of package metadata indexed by package
  * name.
  *
- * "simple" corresponds to the {@link PackageVersionList} interface. It contains a map of package names to versions.
+ * "simple" corresponds to the {@link PackageRange} interface. It contains a map of package names to versions.
  *
- * "caret" corresponds to the {@link PackageCaretRange} interface. It contains a map of package names to
+ * "caret" corresponds to the {@link PackageRange} interface. It contains a map of package names to
  * caret-equivalent version range strings.
  *
- * "tilde" corresponds to the {@link PackageTildeRange} interface. It contains a map of package names to
+ * "tilde" corresponds to the {@link PackageRange} interface. It contains a map of package names to
  * tilde-equivalent version range strings.
+ *
+ * "legacy-compat" corresponds to the {@link PackageRange} interface. It contains a map of package names to
+ * legacy compat equivalent version range strings
  */
-export type ReportKind = "full" | "caret" | "tilde" | "simple";
+export type ReportKind = "full" | "caret" | "tilde" | "simple" | "legacy-compat";
 
 /**
  * Converts a {@link ReleaseReport} into different formats based on the kind.
  */
 export function toReportKind(
 	report: ReleaseReport,
 	kind: ReportKind,
-): ReleaseReport | PackageVersionList | PackageTildeRange | PackageCaretRange {
-	const toReturn: PackageVersionList | PackageTildeRange | PackageCaretRange = {};
+): ReleaseReport | PackageRange {
+	const toReturn: PackageRange = {};
 
 	switch (kind) {
 		case "full": {
+			for (const [, details] of Object.entries(report)) {
+				if (details.releaseGroup !== "client") {
+					details.ranges.legacyCompat = details.ranges.caret;
+				}
+			}
```

Comment:
I thought we landed on only applying this logic to the legacy-compat report? Maybe you and Jason discussed more. If this is needed in more than one place, pull it out into a single function that takes the report details (or whatever the minimum data needed is) and returns the right range. That way the "special code" is in one place, so if it needs to change, it's obvious where. That also gives you a place to document why the special logic exists (hard to document when you multiple instance of the code. Do you doc the first? All of them? Etc.)